.github/workflows/pr-smoke-testing.yaml

```
@@ -12,10 +12,18 @@ permissions:
-  kind-cluster:
+  smoke:
+    strategy:
+      matrix:
+        config: ['private-images', 'public-images']
+      # Allow both matrix jobs to run to completion even if one fails
+      fail-fast: false
     runs-on: "ubuntu-24.04"
     steps:
+      - name: Set environment
+        if: ${{ matrix.config == 'private-images' }}
+        run: echo "ENVIRONMENT=private-images" >> $GITHUB_ENV
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:1:38: Double quote to prevent globbing and word splitting

(shellcheck)